src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -1528,15 +1528,129 @@ groupsBy :: Monad m
     -> Fold m a b
     -> Stream m a
     -> Stream m b
-groupsBy cmp f = foldMany (FL.groupBy cmp f)
+groupsBy cmp f (Stream step state) = Stream (stepOuter f) (Just state, Nothing)
+
+    where
+
+    {-# INLINE_LATE stepOuter #-}
+    stepOuter (Fold fstep initial done) gst (Just st, Nothing) = do
+        res <- step (adaptState gst) st
+        case res of
+            Yield x s -> do
+                fs <- initial
+                sfs <- fstep fs x
+                case sfs of
+                    FL.Partial fs1 -> go SPEC x s fs1
+                    FL.Done fb -> return $ Yield fb (Just s, Just x)
+
+            Skip s    -> return $ Skip $ (Just s, Nothing)
+            Stop      -> return Stop
+
+        where
+
+        go !_ prev stt !acc = do
+            res <- step (adaptState gst) stt
+            case res of
+                Yield x s -> do
+                    if cmp x prev
+                    then do
+                        sfs <- fstep acc x
+                        case sfs of
+                            FL.Partial fs1 -> go SPEC prev s fs1
+                            FL.Done fb -> return $ Yield fb (Just s, Just x)
+                    else done acc >>= \r -> return $ Yield r (Just s, Just x)
+                Skip s -> go SPEC prev s acc
+                Stop -> done acc >>= \r -> return $ Yield r (Nothing, Nothing)
+
+    stepOuter (Fold fstep initial done) gst (Just st, Just prev) = do
+        fs <- initial
+        sfs <- fstep fs prev
+        case sfs of
+            FL.Partial fs1 -> go SPEC st fs1
+            FL.Done fb -> return $ Yield fb (Just st, Nothing)
+
+        where
+
+        -- XXX code duplicated from the previous equation
+        go !_ stt !acc = do
+            res <- step (adaptState gst) stt
+            case res of
+                Yield x s -> do
+                    if cmp x prev
+                    then do
+                        sfs <- fstep acc x
+                        case sfs of
+                            FL.Partial fs1 -> go SPEC s fs1
+                            FL.Done fb -> return $ Yield fb (Just s, Just x)
+                    else done acc >>= \r -> return $ Yield r (Just s, Just x)
+                Skip s -> go SPEC s acc
+                Stop -> done acc >>= \r -> return $ Yield r (Nothing, Nothing)
+
+    stepOuter _ _ (Nothing,_) = return Stop
 
 {-# INLINE_NORMAL groupsRollingBy #-}
 groupsRollingBy :: Monad m
     => (a -> a -> Bool)
     -> Fold m a b
     -> Stream m a
     -> Stream m b
-groupsRollingBy cmp f = foldMany (FL.groupByRolling cmp f)
+groupsRollingBy cmp f (Stream step state) =
```

Comment:
We can define it using parseMany.

Reply:
We will do it in a future change. The parser functions are still not defined I believe.